subsys/bluetooth/mesh/provisioner.c

```diff
@@ -667,6 +669,8 @@ static void prov_set_method(uint8_t method, uint8_t action, uint8_t size)
 	bt_mesh_prov_link.oob_method = method;
 	bt_mesh_prov_link.oob_action = action;
 	bt_mesh_prov_link.oob_size = size;
+
+	(void)memset(bt_mesh_prov_link.auth, 0, sizeof(bt_mesh_prov_link.auth));
```

Comment:
If you really want to move this out of set_input, I believe it needs to go in reset_state, or something else that's always called between provisioning sessions. The application doesn't have to call this for every run, which means that the input oob value may have dirty data at the end if you provision multiple devices without setting the oob method every time.